src/transform-applier.js

```diff
@@ -505,14 +505,42 @@ const _parseUrl = (value, windowRef) => {
  */
 const transformStrokeWidths = function (svgTag, windowRef, bboxForTesting) {
     const inherited = Matrix.identity();
-    const applyTransforms = (element, matrix, strokeWidth, fill) => {
+
+    const makeTransformedGradient = (gradientId, element, matrix) => {
+        const doc = windowRef.document;
+        // Need path bounds to transform gradient
+        const svgSpot = doc.createElement('span');
+        let bbox;
+        if (bboxForTesting) {
+            bbox = bboxForTesting;
+        } else {
+            try {
+                doc.body.appendChild(svgSpot);
+                const svg = SvgElement.set(windowRef.document.createElementNS(SvgElement.svg, 'svg'));
+                const path = SvgElement.set(windowRef.document.createElementNS(SvgElement.svg, 'path'));
+                path.setAttribute('d', element.attributes.d.value);
+                svg.appendChild(path);
+                svgSpot.appendChild(svg);
+                // Take the bounding box.
+                bbox = svg.getBBox();
+            } finally {
+                // Always destroy the element, even if, for example, getBBox throws.
+                doc.body.removeChild(svgSpot);
+            }
+        }
+
+        return _createGradient(gradientId, svgTag, bbox, matrix);
+    };
+
+    const applyTransforms = (element, matrix, strokeWidth, fill, stroke) => {
         if (_isContainerElement(element)) {
             // Push fills and stroke width down to leaves
             if (element.attributes['stroke-width']) {
                 strokeWidth = element.attributes['stroke-width'].value;
             }
-            if (element.attributes && element.attributes.fill) {
-                fill = element.attributes.fill.value;
+            if (element.attributes) {
+                if (element.attributes.fill) fill = element.attributes.fill.value;
+                if (element.attributes.stroke) stroke = element.attributes.stroke.value;
```

Comment:
Why are we checking if `element.attributes` exists here but not checking it in the previous if-block (`if (element.attributes['stroke-width'])`)?

Reply:
It looks like element.attributes should exist if element.tagName exists, which is what we check for in _isContainerElement, except on Internet Explorer. So the if above would break on IE (but we don't support IE anyway)